### DISCUSSION TITLE
Preventing minecart hoppers from being used in others claims

### DISCUSSION THREAD

**Discussion by Kronnixle** [GriefPrevention/GriefPrevention#1228] [Ideas]
If possible, have a feature which checks if the minecart hopper belongs to the player's claim, if not, then prevent the hopper from grabbing items from the player.
I was considering disabling all minecart hoppers, however I'm aware that quite a few players use it on the server.
Is it possible to assign a player to the hopper that they placed, then when that hopper attempts to get items from a chest, check if the chest is in the claim land of the person who owns the hopper?

**Comment by bigpresh** [comment]
How are people placing minecarts in another's claim?  As long as you don't have rails right up to the edge of your claim or something silly, then an untrusted player shouldn't be able to get a hopper minecart in & out.

I think checking the "owner" of a minecart and whether it's in a claim on each hopper tick as items transfer would be pretty expensive... and cause confusion too.

**Comment by Kronnixle** [comment]
Although it can be a bit tedious, it's possible to push a minecart over land, which can then be pushed back out of the claim land. It's not something that's a huge problem, I might install lockette just to prevent situations like this from happening (I believe lockette prevents all hopper minecart instances from being prevented).
Awesome plugin by the way, and thanks for the quick response 🙂

________________________________
